java/ql/lib/semmle/code/java/security/SensitiveUiQuery.qll

```diff
@@ -20,3 +21,96 @@ private module NotificationTrackingConfig implements DataFlow::ConfigSig {
 
 /** Taint tracking flow for sensitive data flowing to system notifications. */
 module NotificationTracking = TaintTracking::Global<NotificationTrackingConfig>;
+
+/** A call to a method that sets the text of a `TextView`. */
+private class SetTextCall extends MethodCall {
+  SetTextCall() {
+    this.getMethod()
+        .getAnOverride*()
+        .hasQualifiedName("android.widget", "TextView", ["append", "setText", "setHint"]) and
+    (
+      this.getMethod()
+          .getParameter(0)
+          .getType()
+          .(RefType)
+          .hasQualifiedName("java.lang", "CharSequence")
+      or
+      this.getMethod().getParameter(0).getType().(Array).getElementType() instanceof CharacterType
+    )
+  }
+
+  /** Gets the string argument of this call. */
+  Expr getStringArgument() { result = this.getArgument(0) }
+}
+
+/** A call to a method indicating that the contents of a UI element are safely masked. */
+private class MaskCall extends MethodCall {
+  MaskCall() {
+    this.getMethod().getAnOverride*().hasQualifiedName("android.widget", "TextView", "setInputType")
+    or
+    this.getMethod().getAnOverride*().hasQualifiedName("android.view", "View", "setVisibility")
+  }
+}
```

Comment:
Is it worth checking the arguments of the method call? Either that it's `View.INVISIBLE` or its literal value, or some `InputType.*PASSWORD*` variation, its literal value, or a binary OR operation with one of those as an operand.

Reply:
It's a potential precision improvement to reduce FNs, but I'm not sure it's worthwhile. For `setVisibility`, it seems redundant; as the visibility is either being set from invisible to visible, or from visible to invisible; either implying it would have been invisible at some point. For `setInputType`, ignoring it yields a few more results in just one project, which set input type to values other than passwords so would appear with a more precise check here, but i'm not sure whether they are TPs.

Reply:
Ok, let's just roll with it and revisit this in the future if we start seeing FNs.